### PR TITLE
Week 3 nic last

### PR DESCRIPTION
**Pull request by niclast7611**
# Submitting Week 2 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [x] Coding Challenges
- [x] Interview Questions
- [x] React Debugging

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:
It was the toughest code challenges of the assessment yet. I felt super discouraged and defeated that I could not figure out questions one and two in the time limit I gave myself. I wanted to push myself like Austin recommended with a time limit. If I didn't have a time limit i would've spent all weekend trying to figure it out.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,42 +21,154 @@
...
+
+const fibo = (arrNum) => {
+    for(let i = 2; i < arrNum; i++) {
+      arrNum[i] = arrNum[i-2] + arrNum[i-1];
+    return arrNum[i];
+    }
+     
+}
+//COULD NOT FIGURE OUT IN TIME LIMIT****
+
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:66]: Thanks for giving it a solid try! In the future lay out what you would have looked into next so you are practicing the art of thinking ahead. 

You probably realize it by now but line 60 requires an array to already be made that you are calling upon to look at the different indices.

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,42 +21,154 @@
...
+const oddOrdered = (array) => {
+    return array.filter(value => {
+       let num = typeof value === 'number'
+       return num.sort((a,b) => a-b)
+   
+       })
+}
+
+//COULD NOT FIGURE OUT IN TIME LIMIT ***
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:114]: What you have here is really solid syntax for a .forEach(). Filter is looking for boolean values to return so it cannot sort trues and falses. Filter needs to work with the values giving you numbers and after that you want to sort all of the numbers into their order.

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,42 +21,154 @@
...
+
+// FAIL  ./code-challenges.test.js
+// fibo
+//   ✕ returns an array that length containing the numbers of the Fibonacci sequence. (3 ms)
+// oddOrdered
+//   ✕ returns a new array of only odd numbers sorted from least to greatest.
+// accuSum
+//   ✓ returns an array of the accumulating sum. An empty array should return an empty array. (1 ms)
+
+//IT SAYS FAIL BECAUSE I COULDNT FIGURE OUT THE FIRST TWO PROBLEMS BUT accuSum IS PASSED
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:173]: Boo ya!

File: interview-questions.md [hunk 4]
```diff
@@ -6,62 +6,50 @@ First, without external resources. Challenge yourself to answer from memory as i
...
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: JSX is a language we use in React it is mainly a combination of HTML and JavaScript. Even though it is a combination there are some syntax differences when using it. For example when you are assigning a method, variable, or pretty much anything instead of using "", we use {} instead. I have used JSX whenever I am inside a React application. I have specifically used the different syntax variations when I assigned an onClick to a function.
 
+Researched answer: JSX stands for Javascript XML, it allows devs to write HTML in React, it also makes it easier to add HTML in React. One main difference from HTML and JSX is that in JSX you can only return on thing and in order to do that we wrap everything we want to display in either a div or a React fragment.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:31]: Well done!

File: interview-questions.md [hunk 5]
```diff
@@ -6,62 +6,50 @@ First, without external resources. Challenge yourself to answer from memory as i
...
+Researched answer: Yarn is a package manager for code. It allows devs to use and share code with other devs all over the world quickly, securely, and reliably.
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
-
-  Researched answer:
+Your answer: I dont think I have ever expierenced or used an anonymous function but a normal function is a reusable piece of code that we can invoke whenever wherever, significatly reducing the amount of typing the dev has to do. If I were to guess what an anonymous function is I would guess it is a function with no name.
 
+Researched answer: I was suprisingly right, an anonymous function is a function without a name so when the dev is creating the function instead of doing const function name and then arrow syntax, you just do const and arrow syntax. However with an anonymous function you cannot call it after creation because it does not have a name to call.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:43]: A good example of an anonymous function is the call back function we pass to map(()=>{}) filter(()=>{})  or sort(()=>{})

File: react-debugging.html [hunk 6]
```diff
@@ -62,12 +62,18 @@
           }
           // No need to change anything ABOVE this line ^
 
+
+          //The first problem I found was that the return needed to be wrapped in React fragments or a div so I added fragments
+          //Second problem was the first button onClick did not have anything in the curly brackets so nothing wouldve happened when the button was clicked so i added the this.drawCard method which was created above allowing the user to click a button and draw a card until there were no more to draw. You use this. because your refering to the document you are currently working in.
+          // The third problem was that the second buttons onClick did not have this. and we use it because of what I mentioned above. Without it the onClick does not know what the method is we are calling.
```
Comment (WalkerTexas-coder) [inline react-debugging.html:68]: Excellent work!

**Review by WalkerTexas-coder — APPROVED** [review]
Nic you crushed it! Great work keeping to your time frames and trying your best. The only feedback I have is to write out your best guess at the next steps when you get stuck so that you are communicating with your future self and other devs which way you were heralded. Approved!

File: code-challenges.test.js (hunk 1 above)
Comment (niclast7611) [reply code-challenges.test.js:66]: Thank you, yes now i realize what i could do different you going over it this morning really helped out.

**Pull request merged**